### DISCUSSION TITLE
adding a new Bitcoin account needs simplifying

### DISCUSSION THREAD

**Discussion by aspiers** [rotki/rotki#8935] [Ideas]
Let's say I'm a newbie user adding a Bitcoin account to Rotki.  The first choice in the dialog is "Manual" or "XPUB".  That's already pretty cryptic for a newbie, so I think there needs to be a quick explanation of how to choose the right one.

That aside, let's say I now choose "XPUB".  Now I see this:

![image](https://github.com/user-attachments/assets/c08ee80e-f57f-4cbb-b071-c09cdbe6f597)

In the first field, I'm expected to choose between P2PKH, P2SH-P2WPKH, WPKH, and P2TR.  Honestly this would be enough for a lot of users to run away screaming!  I'm technical, with some understanding of terms like "segwit", "native segwit", "taproot", and yet even these _slightly_ more human-readable terms are not mentioned.  This seems like a very strange approach to UX, almost as if the app is built primarily for cryptographers and experts, but surely that's not the intention?

Better would be to embed [the helpful hints already in the documentation](https://docs.rotki.com/usage-guides/accounts-and-balances.html#bitcoin-chains):

> - **P2PKH**: This is called legacy. Addresses generated by such an xpub have "1" as a prefix.
> - **P2SH-P2WPKH**: This is called segwit. Addresses generated by such an xpub have "3" as a prefix.
> - **WPKH**: This is called native segwit. Addresses generated by such an xpub have "bc1" as a prefix.
> - **P2TR**: This is called taproot. Addresses generated by such an xpub have "bc1p" as a prefix.

But best of all would be to say "just paste your key in here", do as much auto-detection as possible, and if it's not possible, then display a message guiding the user on how to provide the needed information.

Just in case there's any doubt, while it's awesome that Rotki has thorough docs, this is not a substitute for good UI.  There's a ["Don't Make Me Think"](https://sensible.com/dont-make-me-think/) design principle which means that if your user has to read the docs to understand the UI, your UI design has failed.

**Comment by LefterisJP** [comment]
>  almost as if the app is built primarily for cryptographers and experts, but surely that's not the intention?

The concept of an XPUB and the different xpub types are not something we have invented. Similar UIs exist in other apps that import xpubs.

Embedding those hints in the UI may work as an improvement but autodetection is tricky as those prefixes are just hints and not everyone follows them. Using an XPUB is not easy so the user needs some kind of expertise.

**Reply by aspiers** [reply]
> The concept of an XPUB and the different xpub types are not something we have invented.

Sure, of course :)

> Similar UIs exist in other apps that import xpubs.

Actually my feedback is coming from my experience that some of your rivals make this far friendlier to users.  For instance, Coinstats doesn't mention a single technical term, just asks the user to paste:

![image](https://github.com/user-attachments/assets/3baa4b34-a7f1-4de7-8a93-9249eca59389)

> Embedding those hints in the UI may work as an improvement

They will, trust me :)

> but autodetection is tricky as those prefixes are just hints and not everyone follows them.

Autodetection could include a check whether any balances were found.  If no balance was found, let the user know that something went wrong and that they'll need to provide more info about the key type.  And then guide them through how to find out and provide that info.  Even a deeplink to the relevant docs would be an enormous improvement! 

> Using an XPUB is not easy so the user needs some kind of expertise.

I'd challenge that way of thinking.  If users need this kind of expertise then we won't onboard the next [mbtz]illion users into crypto, or at least not into rotki.  Waiting for users to learn about P2PKH vs. P2TR etc. is unfortunately doomed to fail.  In contrast, a great UI can guide a user into gaining the relevant expertise, rather than just implying "go away and learn stuff, and come back when you're ready".

**Comment by aspiers** [comment]
I just rediscovered https://github.com/rotki/rotki/issues/1664 which was basically about me struggling with exactly the same thing in 2020.  Originally reported on Discord: https://discord.com/channels/657906918408585217/745172610240872508/771720464463822870

**Reply by aspiers** [reply]
After much trial and error, I found I could import the xpub by using https://jlopp.github.io/xpub-converter/ to convert to zpub and then pasting the zpub in the xpub field.  ypub didn't work, however!  I think this is further evidence that the process needs significant simplification.

**Comment by ilovepancakes95** [comment]
I agree that at a minimum those 4 hints on each xpub type should be embedded as a tooltip or something.

**Comment by kelsos** [comment]
We have created an issue to address this: #11693

The plan is to simplify the flow to a single paste field with auto-detection for ypub (SegWit) and zpub (Native SegWit), and a simple human-readable Legacy vs Taproot disambiguation for xpub-prefixed keys. Advanced options (type override, derivation path) will be available behind an expandable section.

Scheduled for 1.43.0.
